libcontainer/rootfs_linux.go

```diff
@@ -152,14 +153,46 @@ func mountToRootfs(m *configs.Mount, rootfs, mountLabel string) error {
 		}
 		return nil
 	case "tmpfs":
+		copyUp := m.Extensions&configs.EXT_COPYUP == configs.EXT_COPYUP
+		tmpDir := ""
 		stat, err := os.Stat(dest)
 		if err != nil {
 			if err := os.MkdirAll(dest, 0755); err != nil {
 				return err
 			}
 		}
-		if err := mountPropagate(m, rootfs, mountLabel); err != nil {
-			return err
+		if copyUp {
+			tmpDirPrefix := os.Getenv("XDG_RUNTIME_DIR")
```

Comment:
I meant read only even before we make it read only and I know we will need more changes to get there. 

Could possibly use some other tmpfs mount if specified in the config before this mount.

Sent from my iPhone

> On May 27, 2016, at 3:48 PM, Michael Crosby <notifications@github.com> wrote:
> 
> In libcontainer/rootfs_linux.go:
> 
> >  		stat, err := os.Stat(dest)
> >  		if err != nil {
> >  			if err := os.MkdirAll(dest, 0755); err != nil {
> >  				return err
> >  			}
> >  		}
> > -		if err := mountPropagate(m, rootfs, mountLabel); err != nil {
> > -			return err
> > +		if copyUp {
> > +			tmpDirPrefix := os.Getenv("XDG_RUNTIME_DIR")
> We don't set it to readonly until after the mounts and after we pivot. Also you can just use /dev/ or /run to do this inside the container.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>